app/scripts/directives/notifications/notificationCounter.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+angular
+  .module('openshiftConsole')
+  .component('notificationCounter', {
+    templateUrl: 'views/directives/notifications/notification-counter.html',
+    bindings: {},
+    controller: [
+      '$routeParams',
+      '$rootScope',
+      'Constants',
+      'DataService',
+      function($routeParams, $rootScope, Constants, DataService) {
+        var ENABLE_EVENT_WATCH = _.get(Constants, 'FEATURE_FLAGS.global_event_watch_for_notification_drawer');
+        var counter = this;
+        var drawerHidden = true;
+
+        var rootScopeWatches = [];
+        // this one is treated separately from the rootScopeWatches as
+        // it may need to be updated outside of the lifecycle of init/destroy
+        var notificationListener;
+
+        var eventsWatcher;
+
+        var deregisterEventsWatch = function() {
+          if(eventsWatcher) {
+            DataService.unwatch(eventsWatcher);
+          }
+        };
+
+        var watchEvents = function(projectName, cb) {
+          if(projectName && ENABLE_EVENT_WATCH) {
+            eventsWatcher = DataService.watch('events', {namespace: projectName}, _.debounce(cb, 50), { skipDigest: true });
+          }
+        };
+
+        var watchNotifications = function(projectName, cb) {
+          if(!projectName) {
+            return;
+          }
+          notificationListener = $rootScope.$on('NotificationsService.onNotificationAdded', cb);
+        };
+
+        var deregisterNotificationListener = function() {
+          notificationListener && notificationListener();
```

Comment:
Will do.  From what I gather, `$on` returns a function that nulls out the listener when you call it (` return function() { namedListeners[indexOf(namedListeners, listener)] = null; };`) but might as well be safe!